server/config/config.go

```diff
@@ -303,7 +303,7 @@ func (c *ServerConfig) SnapDir() string { return filepath.Join(c.MemberDir(), "s
 func (c *ServerConfig) ShouldDiscover() bool { return c.DiscoveryURL != "" }
 
 // ReqTimeout returns timeout for request to finish.
-func (c *ServerConfig) ReqTimeout() time.Duration {
+func (c ServerConfig) ReqTimeout() time.Duration {
```

Comment:
Because I only need this method for interface

Reply:
It isn't necessary to change the receiver type here. You can resolve it by changing [health.go#L184](https://github.com/serathius/etcd/blob/233ca3c31590b2a673cd127f504841c2ca9c1c37/server/etcdserver/api/etcdhttp/health.go#L184) to 

```
	scfg := srv.Config()
	ctx, cancel := context.WithTimeout(context.Background(), scfg.ReqTimeout())
```

Your change here can resolve the "problem" (not a real problem), but it introduces confusion for other developers of etcd.  

Based on golang spec below, methods declared with T (instead of *T) are more adaptive, because they belong to method set of both T and *T.  So if you insist on changing the receiver type from *T to T here, then I'd like to suggest to change all the three method: `ReqTimeout`, `ElectionTimeout` and `PeerDialTimeout` for readability and consistency. 

```
The method set of the corresponding [pointer type](https://go.dev/ref/spec#Pointer_types) *T is the set of all methods declared with receiver *T or T (that is, it also contains the method set of T). 
```

Based on this point, we should change all methods (which do not have the need to change the fields' value in-place) to T.  But from the other perspective, the benefit of keep using *T is that it's more effective, because passing a pointer is more effective than passing the whole struct. So it's a tradeoff.  

So the suggestion here is either keep using *T here or change them for all (only) of the three methods above for readability and consistency.

Reply:
Usually golang can automatically convert the type, in this case converting ServerConfig to *ServerConfig. But srv.Config isn't addressable, so golang can't do the converting. There are many solutions for this "problem", and one of them is to split the `srv.Config.ReqTimeout()` to `scfg := srv.Config(); ... scfg.ReqTimeout() ...`.